src/api/contexts/user.context.tsx

```diff
@@ -49,13 +51,17 @@ export function UserContextProvider(props: PropsWithChildren): JSX.Element {
   async function changeMail(mail: string): Promise<void> {
     if (!user) return; // TODO (Krysh) add real error handling
     setIsUserUpdating(true);
-    return changeUser({ ...user, mail })
+    return changeUser({ ...user, mail }, userLinkAction)
```

Comment:
Why not add the callback as parameter to changeMail?
```ts
async function changeMail(mail: string, onUserLink: () => void): Promise<void> {
```

Reply:
Ah because it is called from different locations?

Reply:
Maybe rename the `register` to something more meaningful then?

Reply:
we need the functionality just once from the main screen. Then all other occurrences of email edit leading to 202 would trigger the popup, which is very neat. As I am not forced to add the same functionality to every change email functionality.

Regarding the name it is registering it once and I wanted to keep it plain as we are using a same naming for metamask hook. What if a second hook is added, would we add a new function or add a new parameter to the `register` function. I would keep it like that

Reply:
Ok agreed 👍